ui/components/MesheryPerformanceComponent.js

```diff
@@ -112,15 +112,22 @@ class MesheryPerformanceComponent extends React.Component {
   }
 
   handleChange = (name) => (event) => {
-    if (name === 'url' && event.target.value !== '') {
-      this.setState({ urlError: false });
+    const value = event.target.value;
+    const pattern = new RegExp('^https?:\\/\\/' + '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + '((\\d{1,3}\\.){3}\\d{1,3}))' + '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + '(\\?[;&a-z\\d%_.~+=-]*)?' + '(\\#[-a-z\\d_]*)?$', 'i');
+
+    if (name === 'url' && value != '') {
+      if (!!pattern.test(value)) {
+        this.setState({ urlError: false });
+      }
+      else {
+        this.setState({ urlError: true });
+      }
     }
-    if (name === 't' && (event.target.value.toLowerCase().endsWith('h')
-      || event.target.value.toLowerCase().endsWith('m') || event.target.value.toLowerCase().endsWith('s'))) {
+    if (name === 't' && (value.toLowerCase().endsWith('h')
```

Comment:
don't do useless assignments, ES-Lint will complain about them.

Reply:
@kushthedude, Please review the changes now!